server/app.js

```diff
@@ -37,4 +39,176 @@ io.on('connection', (socket) => {
+        console.log('emitted');
+        socket.emit('tableState', JSON.stringify(tables[0].getTableState()));
+    }
+}
+function canUserJoinTable(tableToJoin, profile){
+    if(tableToJoin === undefined || tableToJoin.isTableFull() || tableToJoin.isPlayerAtTable(profile) || profile.chips < tableToJoin.buyIn){
+        return false; // the user cannot join this table
+    }
+    else{
+        return true;
+    }
+}
+function joinTable(io, socket, msg){
+    let profile = profiles.find(profile => profile.userID === msg.userID); // get their profile
+    let tableToJoin = tables.find(table => table.tableID === msg.tableID); // get the table they wish to join
+    if(canUserJoinTable(tableToJoin, profile)){// the user can join the table
+        let seatID = tableToJoin.addPlayerToTable(profile, socket.id); // the table will fill a seat based on the profile and send back the seat id
+        socket.emit('yourSeatID', seatID); // send him back his tableSeatID
+        io.emit('tableState', JSON.stringify(tableToJoin.getTableState())); // send to everyone the updated table state
+        if(tableToJoin.canAGameBegin() && !tableToJoin.tableActive){ // check if we can begin a game
+            tableToJoin.tableActive = true; // the table is now active
+            beginTheGame(io, tableToJoin);
+        }
+    }
+    else{
+        return; // the user can not join the table
+    }
+}
+
+
+
+
+
+
+
+
+
+// the functions for communication between the server and the poker.vue page
+function beginTheGame(io, table){ // function will begin the game with a time out of 5 seconds
+    setTimeout(function() { // in five seconds the game will begin
+        console.log('beginning the game');
+        table.beginTheGame(); // the game is set up
+        io.emit('tableState', JSON.stringify(table.getTableState())); // send to everyone the new table state
+        for(let i = 0; i < 6; i++){
+            let seat = table.tableSeats[i];
+            if(seat.socketID !== -1){
+                console.log('yes');
+                io.to(`${seat.socketID}`).emit('beginTheGame', JSON.stringify(seat.cards));
+            }
+        }
+        // now the player whos the current turn can send us a decision
+    }, 5000);
+}
+function turnDecision(io, socket, msg){ // each player when clicking the poker.vue buttons will trigger this function, it will assess whether or not the move was valid
+    if(tables[0].isItTheirTurn(msg.userID, msg.seatID, socket.id)){
+        // there is a limited number of actions a player can take based on their current state
+        // ask if they can make that move, if not then do nothing, later also disable buttons based on lack of options
+        let response = tables[0].playerAction(msg.action, msg.raiseToValue); // will assess the action and return a response
+        if(response){
+            io.emit('tableState', JSON.stringify(tables[0].getTableState()));
+        }
+        else{
+            console.log('bad move');
```

Comment:
Should this be handled at all?

Reply:
bad move just tells me the player tried to do something that was not allowed (i.e. trying to raise when he doesn't have enough chips) in which case the move was ignored

later in development i want to have the buttons enabled/disabled depending on what the server declares allowed at the time of the player's turn. after that we should never get a 'bad move', if that's not feasible then I'll have it send a message back indicating the move was invalid.